### PR TITLE
#90 - Add negative tests for IndexByDirectory

### LINKED ISSUE #90: Obtains `index.yaml` file based on the prefix
It is necessary to provide a method for obtaining `index.yaml` based on the prefix needs to be added.
See: https://helm.sh/docs/helm/helm_repo_index/
Part of #83

### PR DESCRIPTION
**Pull request by genryxy**
Part of #90 
Added negative test cases for `IndexByDirectory`. Used `toString()` in `IndexYaml`

### CODE REVIEW

File: src/test/java/com/artipie/helm/metadata/IndexByDirectoryTest.java [hunk 1]
```diff
@@ -79,6 +82,55 @@ void worksProperlyForManyVersions() {
...
+                (res, exc) -> {
+                    MatcherAssert.assertThat(
+                        Throwables.getRootCause(exc).getMessage(),
+                        new IsEqual<>("Input is not in the .gz format")
+                    );
+                    return Optional.empty();
+                }
+            ).toCompletableFuture()
+            .join();
+    }
```
Comment (olegmoz) [inline src/test/java/com/artipie/helm/metadata/IndexByDirectoryTest.java:132]: @genryxy I'd suggest to capture `Throwable` in `handle` method and move `MatcherAssert.assertThat` to test method body. Assertion code should be executed in any case and it is much more likely to be skipped if it is executed inside a callback function.

**Review by olegmoz — CHANGES_REQUESTED** [review]
@genryxy thanks, please see one comment

**Comment by genryxy** [comment]
@olegmoz thanks, corrected

**Review by olegmoz — APPROVED** [review]
@genryxy thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/helm-adapter#92` is not assigned, can't get performer

**Pull request merged**